pkg/cmd/roachtest/test_runner.go

```diff
@@ -605,17 +605,22 @@ func (r *testRunner) runWorker(
 			qp.Release(alloc)
 		}
 	}()
-
+	// cluster destroy can be done concurrently. We just need to make sure that we exit the roachtest runner once all
+	// the cluster destroy operations are complete!
+	clusterDestroyWg := &sync.WaitGroup{}
+	defer clusterDestroyWg.Wait() // wait for the clusters to be destroyed
 	// Loop until there's no more work in the pool, we get interrupted, or an
 	// error occurs.
 	for {
 		select {
 		case <-interrupt:
 			l.ErrorfCtx(ctx, "worker detected interruption")
+			c.Destroy(context.Background(), closeLogger, l)
```

Comment:
What's the rationale for adding a blocking call on these paths? Note that generally speaking this call has no upper bound on the running time.

Reply:
makes sense. Will remove this and rely of GC to handle this.